Dockerfile

```diff
@@ -0,0 +1,44 @@
+FROM python:3.11-bullseye
+
+LABEL org.opencontainers.image.authors="dev@shoobx.com"
+
+
+RUN apt-get update && apt-get upgrade -y
+
+ARG APP_USER=shoobx\
+    APP_GROUP=shoobx\
+    CODE_FOLDER=/shoobx/shoobx.mocks3 \
+    USER_ID=2000 \
+    GROUP_ID=2000
+
+RUN groupadd --gid $GROUP_ID --non-unique $APP_GROUP && \
+    useradd --no-log-init --uid $USER_ID --non-unique --gid $GROUP_ID --create-home --shell /bin/bash $APP_USER && \
+    echo Created user $USER_ID and group $GROUP_ID
+
+WORKDIR $CODE_FOLDER
+
+RUN mkdir var etc
+
+RUN chown -R $APP_USER:$APP_GROUP $CODE_FOLDER var etc
+
+
+COPY . .
+RUN pip install -r requirements.txt
+
+ENV LOG_LEVEL=INFO \
+    DIRECTORY=./data \
+    HOSTNAME=localhost\
+    RELOAD=True\
+    DEBUG=False\
+    HOST_IP=0.0.0.0 \
+    HOST_PORT=8081 \
+    CORS_ORIGIN=0.0.0.0:8081 \
+    CORS_HEADERS="Origin, ..." \
+    CORS_CREDENTIALS=true \
+    CORS_METHODS="GET, POST, PUT" \
+    CORS_EXPOSE_HEADERS="..."
```

Comment:
Hm, should this really be here? I think this is a docker build environment, that is not actually used during the build (correct me if I'm wrong, but I haven't seen this as a pattern in other projects). Defaults belong to source code. The difference is if someone wants to run the mocks3 outside of docker, mocks3 will not have any of these defaults on.

Reply:
This default values are configured only for docker run. If a user want to run mocks3 outside he may use config which has default values for the run.

`ENV` might be used during build as you mention, but also might be used for running app itself, because they are persistent unlike `ARG` 

Open source examples of `ENV` where they used for configuration of the app:
Selenium documentation [where](https://github.com/seleniumhq-community/docker-seleniarm/tree/trunk#setting-screen-resolution) vars mentioned

Dockerfile [example](https://github.com/seleniumhq-community/docker-seleniarm/blob/trunk/NodeBase/Dockerfile#L165-L186)

One more [doc](https://github.com/seleniumhq-community/docker-seleniarm/tree/trunk#increasing-session-concurrency-per-container)

[Dockerfile](https://github.com/seleniumhq-community/docker-seleniarm/blob/trunk/NodeBase/Dockerfile#L190-L198)

Postgres [doc](https://github.com/docker-library/docs/tree/master/postgres#pgdata)

[Dockerfile](https://github.com/docker-library/postgres/blob/2f0ed0c7e8f8b05b294740f150397eec0af8dc50/16/bookworm/Dockerfile#L180)

Reply:
But doesn't it leave the app completely unconfigured if someone doesn't want to use docker container to run it? This makes It is effectively unusuable without docker. You have to go and set *all* the environment variables yourself in order to do that.

Reply:
When a user type `pip install mocks3` or install it from source code `pip install .` `setup.py` requires to have `config/mocks3.cfg`. This config file will be installed in the same folder `mocks3` installed. 

When a user runs [`sbx-mocks3-serve`](https://github.com/Shoobx/shoobx.mocks3/blob/master/setup.py#L77) (without config path specification) the [default value](https://github.com/Shoobx/shoobx.mocks3/blob/master/src/shoobx/mocks3/run.py#L55) will be taken as result mocks3 will be run with [default configuration](https://github.com/Shoobx/shoobx.mocks3/blob/master/config/mocks3.cfg).

Reply:
Oh, just realized that we now have two configs files in the repo: `config/docker.cfg` and `config/mocks3.cfg`. We use docker.cfg when run via docker and mocks3.cfg when we run via `make run`. 

Well, yeah, this will work but not sure if we really need to split it up and specify same defaults in two places. Adding or removing configuration in this project now requires to do this in two places and making sure they are in sync. You and I will remember this (for a while), no-one else will. Is it worth to introduce such distinction?

These two configs are already diverged, in fact: docker version has cors config, but non-docker doesn't.

Reply:
After experiments I found that `shoobx:cors` section is not used. So there is no section difference between `docker.cfg` and `mocks3.cfg`.

If we want to have ability to adjust container outside though env vars we need to map env vars to config.

I see two solutions:
1) Leave two configs `docker.cfg` filled with env vars and mocks3.cfg which we count as default. And I will add test which will verify that mocks3 is runnable in docker
2) Rewrite `config.py` to have config priority `default -> config -> env var`

I'm ok with any option

Reply:
I will try to rewrite configuration in separate PR and then we can discuss what's best for our case

Reply:
@kedder check https://github.com/Andmedoctopus/shoobx.mocks3/pull/1 please